### DISCUSSION TITLE
Does MVTLayer support 'maskByInstance'?

### DISCUSSION THREAD

**Discussion by lianzhao** [visgl/deck.gl#7576] [Q&A] [answered]
<img width="829" alt="image" src="https://user-images.githubusercontent.com/536786/212310996-d7ec666b-c158-4f65-a8ac-ccfa95c0b3fa.png">

I created two MVTLayers:
1. First one (green layer)
2. Second one (red layer), masked by a circle geoJSON feature with `MaskExtension`.

 I assumed that I could got something like the following result by setting `maskByInstance` to true/false
<img width="297" alt="image" src="https://user-images.githubusercontent.com/536786/212311885-ad0a320f-0ce9-48aa-977a-922fef849630.png">

But the rendered result was not changed. Was I missing something?

**Comment by lianzhao** [comment, marked as answer]
BTW, I'm very interested in the `maskInverted` prop in the master branch. I'd like to implement something like "Turn every building in the circle to red(with some transparency) (maskByInstance=false), and turn other buildings to green (with some transparency)". Will `maskInverted` prop be released in v8.9.x?

**Reply by Pessimistress** [reply]
"Instance" means that the same geometry is being duplicated at various locations, e.g. scatterplot, columns, icons. The documentation of `maskBtyInstance` stated the following:

> controls whether an object is clipped by its anchor (usually defined by an accessor called getPosition, e.g. icon, scatterplot) or by its geometry (e.g. path, polygon). 

Buildings (polygons) by nature are not instanced. There is no such thing as "anchor" for each polygon. So no, the current MaskExtension cannot achieve what you are looking for.

**Reply by lianzhao** [reply]
Thank you for your help.  So may I have some suggestions on how to "Turn the buildings that intersects my 2D polygon to red"? I've tried to read the source code of `SolidPolygonLayer`. I thought I should:
1. Pass uniform values `myPolygon` & `highlightFillColors` to vertex shader.
2. Check if `myPolygon` intersects with `PolygonProps.positions`, update `PolygonProps.fillColors` to my `highlightFillColors` if it is true.

**Reply by lianzhao** [reply]
Haven't figure out how to do it on GPU. I'm using backend PostGIS to calculate intersections instead. Anyway, thank you for your help @Pessimistress .☺️
